### PR TITLE
Added new targets FLYWOOF722PRO, F405HD, updated F745, F405PRO config

### PR DESCRIPTION
**Pull request by flywoo**
1/ Added new target FLYWOOF722PRO: https://flywoo.net/products/goku-f722-pro-v2-30x30-flight-controller-mpu6000-
2/ Added new target FLYWOOF405HD (deleted OSD function based on F405PRO): https://flywoo.net/products/goku-f405-hd-1-2s-elrs-aio-v2
3/ Updated the motor definition of target FLYWOOF745. The early production F745 also has this definition. Added support for ICM42688 and BMI270 gyroscopes
4/ Updated target FLYWOOF405PRO and added config.c file

### CODE REVIEW

**Comment by mmosca** [comment]
This replaces #10049

**Comment by mmosca** [comment]
# FLYWOOF722PRO
 * [x] Flash firmware
 * [x] Calibrate
 * [x] Orientation matches
 * [x] Gyro working
 * [x] Accel working
 * [x] Baro working
 * [ ] Mag working (i2c)
 * [ ] Voltage correct
 * [ ] Current correct
 * [ ] UART1
 * [ ] UART2
 * [ ] UART3
 * [ ] UART4
 * [ ] UART5
 * [ ] UART6
 * [ ] Camera working
 * [ ] Video Out working
 * [ ] OSD working
 * [ ] LEDs working
 * [ ] Buzzer working
 * [ ] Motor outputs
 * [x] Blackbox

# FLYWOOF405HD 
 * [x] Flash firmware
 * [x] Calibrate
 * [x] Orientation matches
 * [x] Gyro working
 * [x] Accel working
 * [x] Baro working
 * [ ] Mag working (i2c)
 * [ ] Voltage correct
 * [ ] Current correct
 * [ ] UART1
 * [ ] UART2
 * [ ] UART3
 * [ ] UART4
 * [ ] UART5
 * [ ] UART6
 * [ ] LEDs working
 * [ ] Buzzer working
 * [ ] Motor outputs
 * [x] Blackbox

File: src/main/target/FLYWOOF722PRO/config.c [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#include "platform.h"
+#include "drivers/pwm_mapping.h"
+#include "fc/fc_msp_box.h"
+#include "io/piniobox.h"
+
+void targetConfiguration(void)
+{
+    pinioBoxConfigMutable()->permanentId[0] = BOX_PERMANENT_ID_USER1;
+	
+    timerOverridesMutable(timer2id(TIM3))->outputMode = OUTPUT_MODE_MOTORS;
```
Comment (mmosca) [inline src/main/target/FLYWOOF722PRO/config.c:27]: These 2 lines can probably be removed.

File: src/main/target/FLYWOOF722PRO/target.c [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+timerHardware_t timerHardware[] = {
+    //DEF_TIM(TIM8,  CH1, PC6,  TIM_USE_PPM,0, 0),  // PPM&SBUS  
+
+    DEF_TIM(TIM3,  CH4, PB1,  TIM_USE_OUTPUT_AUTO, 0, 0),     // S1 - D(1,2)
+    DEF_TIM(TIM3,  CH1, PB4,  TIM_USE_OUTPUT_AUTO, 0, 0),     // S2 - D(1,4)
+    DEF_TIM(TIM2,  CH2, PB3,  TIM_USE_OUTPUT_AUTO, 0, 0),      //  S3 - D(1,6)
+    DEF_TIM(TIM2,  CH1, PA15, TIM_USE_OUTPUT_AUTO, 0, 0),     //  S4 - D(1,5)  
+    DEF_TIM(TIM8,  CH3, PC8,  TIM_USE_OUTPUT_AUTO, 0, 0),      //  S5 - D(2,4) 
+    DEF_TIM(TIM8,  CH4, PC9,  TIM_USE_OUTPUT_AUTO, 0, 0),      //  S6 - D(2,1)
+    DEF_TIM(TIM4, CH1, PB6,   TIM_USE_OUTPUT_AUTO, 0, 0),        //  S7 - D(1,0)
```
Comment (mmosca) [inline src/main/target/FLYWOOF722PRO/target.c:42]: Can you check the dma settings for M7 and M8? Them don't seem to be working with dshot

**Review by mmosca — APPROVED** [review]

**Comment by sensei-hacker** [comment]
The boards listed in the description appear to have four PWM physically present - the motor outputs attached to the onboard ESC.
The target files list 8 PWM outputs, S1 - S8 (plus LED).
What's going on with the four extra outputs?

@flywoo

**Comment by sensei-hacker** [comment]
Screenshot_20241215-101338

Flywoo - is there indeed a PINIO2_PIN?
Or should this be USER1?

**Comment by ProfDrYoMan** [comment]
USER1 is the LED in the prop-guard.

**Pull request merged**